### PR TITLE
feat: add certificate provider config to CLI

### LINKED ISSUE #15: [epic] CLI certificate provider config
### Summary

Add a `certificateProvider` field to the rv1 CLI YAML config file using a discriminated union structure, allowing users to select which TLS certificate provider to use when rendering bundles with webhooks or APIServices. Each provider type can have its own nested configuration.

### Deliverables

- Add `certificateProvider` discriminated union field to `renderConfig` struct in `cmd/rv1/render.go`
- Parse the `type` discriminator and map to library options: `cert-manager` → `CertManagerProvider`, `openshift-service-ca` → `OpenShiftServiceCAProvider`, `none`/empty → no provider (default)
- Design the union struct to support provider-specific config (e.g., `secret` provider will add cert/key data in #19)
- Validate the `type` field — reject unknown values with a clear error message
- Add CLI integration tests: cert-manager config, openshift-service-ca config, none/empty, invalid type
- Update README config file example to show the discriminated union format
- Update `rv1 render --help` long description with cert provider info

### Dependencies

#4

### Scope

small (1-2 files, <100 lines)

### Additional context

The library already exposes `WithCertificateProvider()`, `CertManagerProvider`, and `OpenShiftServiceCAProvider`. This epic is purely CLI config plumbing — no library changes needed.

Config file format using discriminated union:
```yaml
# No provider (default)
certificateProvider:
  type: none

# cert-manager
certificateProvider:
  type: cert-manager

# OpenShift service-ca
certificateProvider:
  type: openshift-service-ca
```

The `type` field is the discriminator. Provider-specific config can be nested (e.g., #19 will add `secret.cert` and `secret.key` fields for the secret provider). This design supports future providers without changing the config schema.

### PR DESCRIPTION
**Pull request by perdasilva**
## Summary
- Add `certificateProvider` discriminated union field to rv1 CLI config file
- `type` discriminator maps to: `cert-manager` → CertManagerProvider, `openshift-service-ca` → OpenShiftServiceCAProvider, `none`/omitted → no provider
- Validate unknown types with clear error listing valid options
- Extensible struct for future provider-specific config (#19 will add `secret` type with cert/key data)
- 6 new integration tests: each provider type, backward compat, invalid type, webhook bundle
- Updated README config example and help text

## Test Plan
- [ ] `make verify` passes
- [ ] `go test -run TestCertificateProvider -v ./cmd/rv1/` — all 6 tests pass
- [ ] `bin/rv1 render --help` shows cert provider in config description
- [ ] Config with `type: bogus` produces error listing valid types
- [ ] All existing regression tests pass unchanged

Closes #15

### CODE REVIEW

File: cmd/rv1/render.go [hunk 1]
```diff
@@ -22,6 +22,25 @@ type renderConfig struct {
...
+	Type string `json:"type"`
+}
+
+var validCertProviderTypes = []string{"none", "cert-manager", "openshift-service-ca"}
+
+func (c *certificateProviderConfig) validate() error {
+	if c == nil || c.Type == "" || c.Type == "none" {
+		return nil
+	}
+	for _, valid := range validCertProviderTypes {
```
Comment (perdasilva) [inline cmd/rv1/render.go]: maybe use the slices library instead? slices.Contains

File: cmd/rv1/render.go [hunk 2]
```diff
@@ -119,6 +147,14 @@ func buildRenderOptions(cfg renderConfig) []regv1render.Option {
 	if cfg.DeploymentConfig != nil {
 		opts = append(opts, regv1render.WithDeploymentConfig(cfg.DeploymentConfig))
 	}
+	if p := cfg.CertificateProvider; p != nil {
+		switch p.Type {
+		case "cert-manager":
```
Comment (perdasilva) [inline cmd/rv1/render.go]: let's make these values constants
